src/main/java/org/embulk/output/SnowflakeOutputPlugin.java

```diff
@@ -130,25 +135,39 @@ protected JdbcOutputConnector getConnector(PluginTask task, boolean retryableMet
   }
 
   @Override
-  public ConfigDiff resume(
-      TaskSource taskSource, Schema schema, int taskCount, OutputPlugin.Control control) {
-    throw new UnsupportedOperationException("snowflake output plugin does not support resuming");
-  }
-
-  @Override
-  protected void doCommit(JdbcOutputConnection con, PluginTask task, int taskCount)
-      throws SQLException {
-    super.doCommit(con, task, taskCount);
-    SnowflakeOutputConnection snowflakeCon = (SnowflakeOutputConnection) con;
-
+  public ConfigDiff transaction(
+      ConfigSource config, Schema schema, int taskCount, OutputPlugin.Control control) {
+    PluginTask task = CONFIG_MAPPER.map(config, this.getTaskClass());
     SnowflakePluginTask t = (SnowflakePluginTask) task;
-    if (this.stageIdentifier == null) {
-      this.stageIdentifier = StageIdentifierHolder.getStageIdentifier(t);
-    }
+    this.stageIdentifier = StageIdentifierHolder.getStageIdentifier(t);
```

Comment:
It seems that we don't have to define `stageIdentifier` as property. (as `this.stageIdentifier`)
`stageIdentifier` can be local variable.